services/cd-service/pkg/repository/repository.go

```diff
@@ -741,35 +741,45 @@ type Lock struct {
 }
 
 func readLock(fs billy.Filesystem, lockDir string) (*Lock, error) {
-	msg, err := readFile(fs, fs.Join(lockDir, "message"))
-	if err != nil {
-		return nil, err
-	}
-	email, err := readFile(fs, fs.Join(lockDir, "created_by_email"))
-	if err != nil {
-		return nil, err
+	lock := &Lock{}
+
+	if cnt, err := readFile(fs, fs.Join(lockDir, "message")); err != nil {
+		if !os.IsNotExist(err) {
+			return nil, err
```

Comment:
is there any way to identify between the cases
- lock doesn't exist
and
- migration not done yet

Reply:
I don't think this is very necessary. We can't really differentiate between the two errors in any meaningful way. What comes to mind is checking if the old lock is there. but that is not applicable. (the old lock is a file and the new lock is a directory with the same name. )

Reply:
okay, so as of now if migration is not done you'll lose locks. that seems a bit dangerous. Which is the only reason I ask